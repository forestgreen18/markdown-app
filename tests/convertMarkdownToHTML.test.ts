/* eslint-disable @typescript-eslint/no-var-requires */
export {}; // Add this line to treat this file as a module

const fs = require('fs');
const { getTestFilePath } = require('../utils/getTestFilePath');
const { convertMarkdownToHTML } = require('../src/converters/markdownToHtml');

describe('convertMarkdownToHTML', () => {
  it('converts valid markdown to HTML', () => {
    const mdFilePath = getTestFilePath('validMD.md');
    const mdContent = fs.readFileSync(mdFilePath, 'utf8');

    const htmlFilePath = getTestFilePath('expected.html');
    const htmlContent = fs.readFileSync(htmlFilePath, 'utf8');
    const expectedOutput = htmlContent; // the expected HTML output

    expect(convertMarkdownToHTML(mdContent)).toEqual(expectedOutput);
  });

  it('throws an error for invalid markdown syntax', () => {
    const mdFilePath = getTestFilePath('invalidMD.md');
    const mdContent = fs.readFileSync(mdFilePath, 'utf8');
    expect(() =>
      convertMarkdownToHTML(mdContent)
    ).toThrowErrorMatchingSnapshot();
  });
});
