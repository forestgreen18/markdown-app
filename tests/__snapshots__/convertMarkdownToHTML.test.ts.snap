// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertMarkdownToHTML throws an error for invalid markdown syntax 1`] = `
"invalid markdown <details are below>
{
  "**_invalid syntax_**": "Embedding styles within each other is prohibited"
}"
`;
